x-pack/plugins/security_solution/cypress/objects/rule.ts

```diff
@@ -359,9 +363,10 @@ export const getNewThresholdRule = (): ThresholdRule => ({
   lookBack: getLookBack(),
   timeline: getTimeline(),
   maxSignals: 100,
+  ...rewrites,
 });
 
-export const getNewTermsRule = (): NewTermsRule => ({
+export const getNewTermsRule = (rewrites?: Partial<NewTermsRule>): NewTermsRule => ({
```

Comment:
Similar to below, the [other PR](https://github.com/elastic/kibana/pull/151520) will make broad changes to the rule mocks, so we'll get significant conflicts here. Once we have the new schemas in place for these functions, I think it will also be easier to apply rewrites to the mock data using a pattern like `{ ...getNewTermsRule(), rule_id: 'new_rule_id'}`.

Reply:
@xcrzx @marshallmain's suggestion conflicts with your https://github.com/elastic/kibana/pull/150553#discussion_r1115537401. I personally prefer provide rewrites to mock creators like this `getNewTermsRule({ rule_id: 'new_rule_id' })` but it will be good to find common ground.

Reply:
I would prefer passing overrides as an argument to the mock builder as it makes the code appear more organized and provides greater flexibility. For instance, we could effortlessly implement a deep merge of the mocked structure with the provided override. 

For example, overriding nested properties become tiresome quickly, as shown in the following code snippet:

```ts
const rule = {
  ...getNewTermsRule(),
  rule_id: 'new_rule_id',
  runsEvery: {
    interval: '1',
    ...getNewTermsRule().runsEvery,
  },
};
```

If we implement deep merge, the readability could be greatly improved:

```ts
const rule = getNewTermsRule({
  rule_id: 'new_rule_id',
  runsEvery: {
    interval: '1',
  },
});
```